src/RulesDeploy.ts

```diff
@@ -53,72 +58,96 @@ export class RulesDeploy {
    * Compile all rulesets tied to this deploy, rejecting on first
    * compilation error.
    */
-  compile(): Promise<any> {
-    const promises: any[] = [];
-    _.forEach(this.rulesFiles, (files: RulesetFile[], filename: string) => {
-      promises.push(this._compileRuleset(filename, files));
-    });
-    return Promise.all(promises);
+  async compile(): Promise<void> {
+    for (const filename of Object.keys(this.rulesFiles)) {
+      const files = this.rulesFiles[filename];
+      await this._compileRuleset(filename, files);
```

Comment:
this compiles rules sequentially instead of in parallel, is that desired?

Reply:
since JS is single threaded, there's no difference (in theory).

This is easier to understand, at any rate, than putting a bunch of promises into an array and awaiting promise.all (though that does have its cases)

Reply:
In this case it's an async network request, so the number of local threads shouldn't matter.

Consider the difference between
```
for (let i = 0; i < 10; i++) {
  await new Promise(r => setTimeout(r, 1000 * i));
}
```
and
```
const ps = [];
for (let i = 0; i < 10; i++) {
  ps.push(new Promise(r => setTimeout(r, 1000 * i)));
}
await Promise.all(ps);
```

Reply:
mmmm. I'm not 100% sure what I was thinking, but your point is correct. I've refactored this again.